server/src/main/java/org/opensearch/search/query_group/QueryGroupServiceSettings.java

```diff
@@ -0,0 +1,256 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.search.query_group;
+
+import org.opensearch.common.settings.ClusterSettings;
+import org.opensearch.common.settings.Setting;
+import org.opensearch.common.settings.Settings;
+
+/**
+ * Main class to declare the QueryGroup feature related settings
+ */
+public class QueryGroupServiceSettings {
+    private static final Double DEFAULT_NODE_LEVEL_MEMORY_REJECTION_THRESHOLD = 0.8;
+    private static final Double DEFAULT_NODE_LEVEL_MEMORY_CANCELLATION_THRESHOLD = 0.9;
+    private static final Double DEFAULT_NODE_LEVEL_CPU_REJECTION_THRESHOLD = 0.8;
+    private static final Double DEFAULT_NODE_LEVEL_CPU_CANCELLATION_THRESHOLD = 0.9;
+    public static final double NODE_LEVEL_MEMORY_CANCELLATION_THRESHOLD_MAX_VALUE = 0.95;
+    public static final double NODE_LEVEL_MEMORY_REJECTION_THRESHOLD_MAX_VALUE = 0.9;
+    public static final double NODE_LEVEL_CPU_CANCELLATION_THRESHOLD_MAX_VALUE = 0.95;
+    public static final double NODE_LEVEL_CPU_REJECTION_THRESHOLD_MAX_VALUE = 0.9;
```

Comment:
Should we have separate MAX values for cancellation and rejection? As a customer, I might want to start cancelling instead of rejecting. So, probably no need to have separate rejection threshold max value

Reply:
The intention behind this is to favor the resiliency. If we favor cancellation over rejection then we will be throwing away lots of work done on cancelled task. If a query group rejection threshold is breached we are still giving some headroom for the ongoing requests in the query group to complete instead of throwing that away.

Let me know if this seems right ?

Reply:
Yeah, it does make sense. Also, lesser settings exposed to the customers, better it is